### PR TITLE
Fixes for the sim tests and the deposits.

### PR DESCRIPTION
**Pull request by StefanIliev545**
### Why is this change needed?

Taken out of the cross chain PR. Fixes a couple of issues like accounts never being funded and the tests not checking that there have been actual deposits.

### What changes were made as part of this PR:

- Provide a high level list of the changes made
- List key areas for the reviewer 

### :rotating_light: Definition of done :rotating_light:
- [ ] Unit tests added to cover new or changed functionality 
- [ ] Docs pages updated to cover new or changed functionality
- [ ] [Changelog.md](https://github.com/obscuronet/go-obscuro/blob/main/docs/testnet/changelog.md) updated

### CODE REVIEW

**Review by tudor-malene — COMMENTED** [review]
one small comment

File: integration/ethereummock/erc20_contract_lib.go [hunk 1]
```diff
@@ -13,7 +16,20 @@ func (c *contractLib) CreateDepositTx(tx *ethadapter.L1DepositTx, nonce uint64)
 }
 
 func (c *contractLib) DecodeTx(tx *types.Transaction) ethadapter.L1Transaction {
```
[the remark this answers is not in the document]
Comment (StefanIliev545) [reply integration/ethereummock/erc20_contract_lib.go:18]: Good call, it's the name that gave me a lot of headaches when I was searching for the issues and I still forgot to change it 😆

**Review by tudor-malene — APPROVED** [review]
lgtm

**Comment by StefanIliev545** [comment]
Closing this as the fixes have been merged with the cross chain messaging PRs

**Pull request closed without merging**